### PR TITLE
use render_in_context for eval form block

### PR DESCRIPTION
**Pull request by Fivell**
https://github.com/activeadmin/activeadmin/pull/3503

### CODE REVIEW

**Comment by Fivell** [comment]
@timoschilling , can you review logic before I start playing with houndci  ? Thanks

File: lib/active_admin/batch_actions/controller.rb [hunk 1]
```diff
@@ -7,7 +7,8 @@ def batch_action
         if action_present?
           selection =            params[:collection_selection] ||  []
           inputs    = JSON.parse params[:batch_action_inputs]  || '{}'
-          inputs    = inputs.with_indifferent_access.slice *instance_exec(&current_batch_action.inputs).keys
+          valid_keys = render_in_context(self, current_batch_action.inputs).try(:keys)
+          inputs    = inputs.with_indifferent_access.slice *valid_keys
```
Comment (timoschilling) [inline lib/active_admin/batch_actions/controller.rb]: bring all line into one column

**Comment by timoschilling** [comment]
look's good, don't worry about the line length comments from @houndci

**Comment by Fivell** [comment]
>>bring all line into one column 
@timoschilling , I hope I understood you.

File: lib/active_admin/batch_actions/controller.rb [hunk 2]
```diff
@@ -5,9 +5,10 @@ module Controller
       # Controller action that is called when submitting the batch action form
       def batch_action
         if action_present?
-          selection =            params[:collection_selection] ||  []
-          inputs    = JSON.parse params[:batch_action_inputs]  || '{}'
-          inputs    = inputs.with_indifferent_access.slice *instance_exec(&current_batch_action.inputs).keys
+          selection  =            params[:collection_selection] ||  []
+          inputs     = JSON.parse params[:batch_action_inputs]  || '{}'
```
Comment (Fivell) [inline lib/active_admin/batch_actions/controller.rb]: @timoschilling ,  maybe we should use single_quotes style ? 
like https://gist.github.com/salbertson/05e579933807493cf1c5

Comment (timoschilling) [inline lib/active_admin/batch_actions/controller.rb]: I know there is a minimal performance difference between single and double quotes. But I prefer option B from the https://github.com/bbatsov/ruby-style-guide#consistent-string-literals, while this style can used for each case, with single quotes some times you need to switch to double quotes

**Comment by timoschilling** [comment]
:+1:

**Pull request merged**